### DISCUSSION TITLE
Draft % Boost

### DISCUSSION THREAD

**Discussion by jeffwink** [SauceLLC/sauce4zwift#51] [General]
I saw your doc that you don't know how the engine uses this, just wanted to confirm that Zwift reports this as a "boost %". Just trying to figure out what might be going on under the hood. Thanks!

Cool project. Happy to support!

**Comment by mayfield** [comment]
~~Hi @jeffwink, thanks for the confirmation.~~  Let us know if you have any other insights into this figure.  There have been a TON of discussions about this on the Sauce discord with several people running case studies to try and map it to a formula.

**Comment by jeffwink** [comment]
Sorry I didn’t do a good job with punctuation, I meant it more as a question.

In any case, I read through the Discord channels and skimmed your source code on GitHub. 

So, Zwift reports this as an int32. Are you doing anything to the value or just displaying it as is? It seems to go from zero to above 300? (I haven’t looked for max values specifically but that could give some insights).

 
message PlayerState {
	...
    int32 draft = 10; // seems to be in percentage / algo factor

**Reply by mayfield** [reply]
Hi @jeffwink,

That's correct about the type and value.  I do literally no alterations of the value from the data stream.  I put a % on it in the UI to just indicate that it's **a** factor in the game's models.  What that factor is and how it's used is an open question.

It's also probably affected by modalities of game play like, doubledraft and the new pack-dynamics-v4 (pd4).  I've even poked around in the assembly code a bit to see if I could make sense of it (I couldn't) and noticed some code that might be trying to quantify when you are in a group and use a different model than solo riding.  However it's entirely possible I misunderstood the machine code; it's not like reading the original C++, lol.

Regarding max value. I think someone on the discord server reported a 1600%!  There is a bragging rights contest to try and hit the biggest number.  In short though, it's similar to IRL in that for a huge number you need to be going super fast in a big group, i.e. tight pack going down Alpe Du' Zwift full gas.
